security/entity_provider.rst

```diff
@@ -140,6 +140,40 @@ For this entry, suppose that you already have a ``User`` entity inside an
 To make things shorter, some of the getter and setter methods aren't shown.
 But you can generate these manually or with your own IDE.
 
+.. tip::
+    Since Symfony 3.4, the User class should implements the ``EquatableInterface``in order to allow 
+    the security to check if the User stored in the Session is the one who's been authenticated, in 
+    order to allow this check, just implement the ``isEqualTo`` method::
+    
+    // src/AppBundle/Entity/User.php
+    namespace AppBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Symfony\Component\Security\Core\User\UserInterface;
+    use Symfony\Component\Security\Core\User\EquatableInterface;
+
+    /**
+     * @ORM\Table(name="app_users")
+     * @ORM\Entity(repositoryClass="AppBundle\Repository\UserRepository")
+     */
+    class User implements UserInterface, EquatableInterface, \Serializable
+    {
+        // ...
+        
+        public function isEqualTo(UserInterface $user) 
+        {
+            if ($this->username !== $user->getUsername() {
```

Comment:
```php
return $this->username === $user->getUsername();
```

?

Reply:
Yep, work fine, I wasn't sure about the capacity of the security layer to recognize the boolean after the condition.